src/main/java/com/linkedin/kmf/services/ConsumeService.java

```diff
@@ -105,25 +104,26 @@ public ConsumeService(Map<String, Object> props, String name) throws Exception {
 
     // Assign config specified for consumer. This has the highest priority.
     consumerProps.putAll(consumerPropsOverride);
-
     _consumer = (KMBaseConsumer) Class.forName(consumerClassName).getConstructor(String.class, Properties.class).newInstance(topic, consumerProps);
-    _thread = new Thread(() -> {
-      try {
-        consume();
-      } catch (Exception e) {
-        LOG.error(_name + "/ConsumeService failed", e);
-      }
-    }, _name + " consume-service");
-    _thread.setDaemon(true);
-
-    MetricConfig metricConfig = new MetricConfig().samples(60).timeWindow(1000, TimeUnit.MILLISECONDS);
-    List<MetricsReporter> reporters = new ArrayList<>();
-    reporters.add(new JmxReporter(JMX_PREFIX));
-    Metrics metrics = new Metrics(metricConfig, reporters, new SystemTime());
-    Map<String, String> tags = new HashMap<>();
-    tags.put("name", _name);
-    _adminClient = AdminClient.create(props);
-    _sensors = new ConsumeMetrics(metrics, tags, topic);
+    topicPartitionReady.thenRun(() -> {
```

Comment:
It is hard to the understand flow of future completions and its consequences. This constructor sneakily chains a runnable to `topicPartitionReady` future _inside_ the constructor. It's much better to refactor the _action_ from _chaining_. One possibility is as follows.
1. Separate out this large lambda in a method of its own.
2. Don't pass the future in constructor. 
3. When the top-level future is created in `MultiClusterTopicManagementService` (ideally not in the constructor), pass a small lambda calling the method extracted in step 1. It's **much** better to chain futures such that all chains (if possible) are visible in one small piece of code.

Reply:
That's correct. I was debating why the original code has a constructor of huge chunk that contains a large lambda. In the separate PR, we will factor out this large lambda.